src/framework/ui/uitypes.h

```diff
@@ -187,47 +189,89 @@ enum class Checkable {
     std::string scCtx = "any";
     TranslatableString title;
     TranslatableString description;
+    ActionCategory category = ActionCategory::Internal;
     IconCode::Code iconCode = IconCode::Code::NONE;
     Checkable checkable = Checkable::No;
     std::vector<std::string> shortcuts;
 
     UiAction() = default;
-    UiAction(const actions::ActionCode& code, UiContext ctx, std::string scCtx, Checkable ch = Checkable::No)
-        : code(code), uiCtx(ctx), scCtx(scCtx), checkable(ch) {}
+    UiAction(const actions::ActionCode& code, ActionCategory cat, UiContext ctx, std::string scCtx, Checkable ch = Checkable::No)
+        : code(code), category(cat), uiCtx(ctx), scCtx(scCtx), checkable(ch) {}
 
-    UiAction(const actions::ActionCode& code, UiContext ctx, std::string scCtx, const TranslatableString& title,
+    UiAction(const actions::ActionCode& code, ActionCategory cat, UiContext ctx, std::string scCtx, const TranslatableString& title,
              Checkable ch = Checkable::No)
-        : code(code), uiCtx(ctx), scCtx(scCtx), title(title), checkable(ch) {}
+        : code(code), category(cat), uiCtx(ctx), scCtx(scCtx), title(title), checkable(ch) {}
 
-    UiAction(const actions::ActionCode& code, UiContext ctx, std::string scCtx, const TranslatableString& title,
+    UiAction(const actions::ActionCode& code, ActionCategory cat, UiContext ctx, std::string scCtx, const TranslatableString& title,
              const TranslatableString& desc, Checkable ch = Checkable::No)
-        : code(code), uiCtx(ctx), scCtx(scCtx), title(title), description(desc),  checkable(ch) {}
+        : code(code), category(cat), uiCtx(ctx), scCtx(scCtx), title(title), description(desc),  checkable(ch) {}
 
-    UiAction(const actions::ActionCode& code, UiContext ctx, std::string scCtx, const TranslatableString& title,
+    UiAction(const actions::ActionCode& code, ActionCategory cat, UiContext ctx, std::string scCtx, const TranslatableString& title,
              const TranslatableString& desc, IconCode::Code icon, Checkable ch = Checkable::No)
-        : code(code), uiCtx(ctx), scCtx(scCtx), title(title), description(desc), iconCode(icon), checkable(ch) {}
+        : code(code), category(cat), uiCtx(ctx), scCtx(scCtx), title(title), description(desc), iconCode(icon), checkable(ch) {}
 
-    UiAction(const actions::ActionCode& code, UiContext ctx, std::string scCtx, const TranslatableString& title, IconCode::Code icon,
+    UiAction(const actions::ActionCode& code, ActionCategory cat, UiContext ctx, std::string scCtx, const TranslatableString& title,
+             IconCode::Code icon,
              Checkable ch = Checkable::No)
-        : code(code), uiCtx(ctx), scCtx(scCtx), title(title), iconCode(icon), checkable(ch) {}
+        : code(code), category(cat), uiCtx(ctx), scCtx(scCtx), title(title), iconCode(icon), checkable(ch) {}
 
     bool isValid() const
     {
         return !code.empty();
     }
 
+    QString getCategory() const
```

Comment:
I would call this function `categoryName()`.  But I'm not sure it really belongs here. It should probably be defined more like this:
```c++
QString name(ActionCategory cat) {
    // code
}
```
And live outside the UiAction struct. It should be declared in a `.h` file and defined in a `.cpp` file.